crates/sui-types/src/storage/mod.rs

```diff
@@ -340,6 +341,51 @@ pub fn get_module_by_id<S: BackingPackageStore>(
         .map(|bytes| CompiledModule::deserialize_with_defaults(&bytes).unwrap()))
 }
 
+/// A `BackingPackageStore` that resolves packages from a backing store, but also includes any
+/// packages that were published in the current transaction execution. This can be used to resolve
+/// Move modules right after transaction execution, but newly published packages have not yet been
+/// committed to the backing store on a fullnode.
+pub struct PostExecutionPackageResolver {
+    backing_store: Arc<dyn BackingPackageStore>,
+    new_packages: BTreeMap<ObjectID, PackageObject>,
+}
+
+impl PostExecutionPackageResolver {
+    pub fn new(
+        backing_store: Arc<dyn BackingPackageStore>,
+        effects: &TransactionEffects,
+        output_objects: &Option<Vec<Object>>,
+    ) -> Self {
+        let object_map: BTreeMap<_, _> = output_objects
+            .as_ref()
+            .map(|objects| objects.iter().map(|obj| (obj.id(), obj)).collect())
+            .unwrap_or_default();
+        let new_packages = effects
+            .published_package_ids_v2_only()
```

Comment:
Instead of introducing this new API, can we introspect the `objects`, to look for the objects that are packages? It seems like a less invasive change.

Reply:
Ah that's a good one. Thanks!